morph_tool/nrnhines.py

```diff
@@ -273,3 +278,87 @@ def point_to_section_end(sections: Sequence[neuron.nrn.Section],
         if np.isclose(point, last_section_point, atol=atol, rtol=rtol).all():
             return index
     return None
+
+
+class NoDaemonProcess(multiprocessing.Process):
+    """Class that represents a non-daemon process"""
+
+    def _get_daemon(self):  # pylint: disable=R0201
+        """Get daemon flag"""
+        return False
+
+    def _set_daemon(self, value):
+        """Set daemon flag"""
+
+    daemon = property(_get_daemon, _set_daemon)
+
+
+class NestedPool(multiprocessing.pool.Pool):  # pylint: disable=abstract-method
+    """Class that represents a MultiProcessing nested pool"""
+
+    Process = NoDaemonProcess
+
+
+def isolate(func):
+    """Isolate a generic function for independent NEURON instances.
+
+    Note: it does not work as decorator.
+
+    Args:
+        func (function): function to isolate"""
+    def func_isolated(*args, **kwargs):
+        with NestedPool(1, maxtasksperchild=1) as pool:
+            return pool.apply(func, args, kwargs)
+    return func_isolated
+
+
+def convert_point_to_isec(morphology_path, point, section_type="apical"):
+    """Convert a point position to NEURON section index and return cell name and id.
+
+    Args:
+        morphology_path (Path): path to morphology
+        point (list/ndarray): 3d point
+    """
+    cell = get_NRN_cell(morphology_path)
+    point = point_to_section_end(getattr(cell.icell, section_type), point)
+    return morphology_path.stem, point
+
+
+def convert_point_to_isec_isolated(morph_data, section_type="apical"):
+    """Serves as a decorator to isolate convert_point_to_isc, but work with multiprocessing."""
+    return isolate(convert_point_to_isec)(*morph_data, section_type=section_type)
```

Comment:
Why is the signature changing from `morph_data` to `*morph_data` during the isolation

Reply:
because it is converted from tuple (coming from the dict.items()) to two entries for the function